5-AccessControl/1-call-api-roles/AppCreationScripts/Configure.ps1

```diff
@@ -0,0 +1,396 @@
+   $appRoles.Add($newRole)
+   Set-AzureADApplication -ObjectId $clientAadApplication.ObjectId -AppRoles $appRoles
+
+   Write-Host "Done creating the client application (msal-react-spa)"
+
+   # URL of the AAD application in the Azure portal
+   # Future? $clientPortalUrl = "https://portal.azure.com/#@"+$tenantName+"/blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Overview/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
+   $clientPortalUrl = "https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/CallAnAPI/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
+   Add-Content -Value "<tr><td>client</td><td>$currentAppId</td><td><a href='$clientPortalUrl'>msal-react-spa</a></td></tr>" -Path createdApps.html
+
+   $requiredResourcesAccess = New-Object System.Collections.Generic.List[Microsoft.Open.AzureAD.Model.RequiredResourceAccess]
+
+   # Add Required Resources Access (from 'client' to 'service')
+   Write-Host "Getting access from 'client' to 'service'"
+   $requiredPermissions = GetRequiredPermissions -applicationDisplayName "msal-node-api" `
+                                                -requiredDelegatedPermissions "access_as_user" `
+
+   $requiredResourcesAccess.Add($requiredPermissions)
+
+
+   Set-AzureADApplication -ObjectId $clientAadApplication.ObjectId -RequiredResourceAccess $requiredResourcesAccess
+   Write-Host "Granted permissions."
+
+   # Update config file for 'service'
+   $configFile = $pwd.Path + "\..\API\authConfig.json"
+   Write-Host "Updating the sample code ($configFile)"
+   $dictionary = @{ "tenantID" = $tenantId;"clientID" = $serviceAadApplication.AppId };
+   UpdateTextFile -configFilePath $configFile -dictionary $dictionary
+
+   # Update config file for 'client'
+   $configFile = $pwd.Path + "\..\SPA\src\authConfig.js"
+   Write-Host "Updating the sample code ($configFile)"
+   $dictionary = @{ "Enter_the_Application_Id_Here" = $clientAadApplication.AppId;"Enter_the_Tenant_Info_Here" = $tenantId;"Enter_the_Web_Api_Scope_here" = ("api://"+$serviceAadApplication.AppId+"/access_as_user") };
+   ReplaceInTextFile -configFilePath $configFile -dictionary $dictionary
+   Write-Host ""
+   Write-Host -ForegroundColor Green "------------------------------------------------------------------------------------------------" 
+   Write-Host "IMPORTANT: Please follow the instructions below to complete a few manual step(s) in the Azure portal":
+   Write-Host "- For 'service'"
+   Write-Host "  - Navigate to '$servicePortalUrl'"
+   Write-Host "  - To receive the `roles` claim with the name of the app roles this user is assigned to, make sure that the user accounts you plan to sign-in to this app is assigned to the app roles of this service instance. The guide, https://aka.ms/userassignmentrequired provides step by step instructions." -ForegroundColor Red 
```

Comment:
Double quotes ("roles") can be used at place of backtick(\`roles\`) because using \`roles\` displays comment as below:
![image](https://user-images.githubusercontent.com/29180994/115311103-ea31ed80-a123-11eb-9e52-a97161b7e710.png)